inc/server/class-prompt-server.php

```diff
@@ -89,15 +110,85 @@ public function register_routes() {
 		);
 	}
 
+	/**
+	 * Save the API key.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return \WP_Error|\WP_HTTP_Response|\WP_REST_Response
+	 */
+	public function save_api_key( $request ) {
+		$body = $request->get_body();
+		$body = json_decode( $body, true );
+
+		if ( ! isset( $body['api_key'] ) ) {
```

Comment:
You want to add an `is_array` check as well because `json_decode` returns `false` if there's an error.